model-server/src/main/kotlin/org/modelix/model/server/store/IgniteStoreClient.kt

```diff
@@ -103,6 +112,44 @@ class IgniteStoreClient(jdbcConfFile: File? = null, inmemory: Boolean = false) :
         return cache.associate { it.key to it.value }
     }
 
+    override fun removeRepositoryObjects(repositoryId: RepositoryId) {
+        if (!inmemory) {
+            // Not all entries are in the cache. We delete them directly instead of loading them into the cache first.
+            // This should be safe as the repository has already been removed from the list of available ones.
```

Comment:
Let's explicitly document this contract in the interface kdoc for `removeRepositoryObjects` a la: "callers need to ensure that repositories are not usable anymore before calling this method". This increases the chance that this fact is not forgotten during the next refactoring.

Reply:
fixed